### DISCUSSION TITLE
Questions about convert-backend and inference

### DISCUSSION THREAD

**Discussion by zhangqi94** [deepmodeling/deepmd-kit#4743] [Q&A]
1. When I use `dp convert-backend model.pth model.jax`, it outputs a checkpoint folder. How can I load it and perform inference with JAX?  
2. I want to use the JAX or PyTorch backend for inference to compute the Hessian, but I couldn’t find relevant documentation about this.  
3. It seems that `DeepPot` relies on NumPy. How can I directly use JAX or PyTorch data types for inference and compute gradients?  

Thank you so much in advance for your help! I really appreciate it if you could provide guidance.

**Comment by njzjz** [comment]
1. The output model extension is designed as `model.savedmodel`. See [documentation](https://docs.deepmodeling.com/projects/deepmd/en/stable/backend.html) for details.
2. We haven't designed a nice API for Hessian, but there is an example in #4668 using the existing API. There are also some codes in the tests.
3. The example in #4668 inputs the JAX data type. It's a good feature request to let `DeepPot` have the ability to accept any array data types (I feel the Array API could be used here).

**Reply by zhangqi94** [reply]
Thank you very much for your response.
Based on your reply, I’ve encountered another issue: when I include `cell` in the JIT-compiled model, it throws an error. However, it works fine without JIT.
`model_call(jnp.array(coord), jnp.array(atype), jnp.array(cell))`
Is there any way to resolve this?
By the way, specifying `cell` as a static argument doesn’t resolve the issue either. `model_call = jax.jit(model.call, static_argnums=(2,))`

**Reply by njzjz** [reply]
call = build neighbost list + call_lower

Building neighbor list for the PBC box cannot use jit since the output shape is dynamic. Call_lower can use jit.
